starknet-foundry/crates/cheatable-starknet/src/rpc.rs

```diff
@@ -474,9 +482,76 @@ impl CheatableSyscallHandler<'_> {
             response_wrapper.write(vm, &mut self.syscall_handler.syscall_ptr)?;
             return Ok(());
         }
+
+        if let SyscallSelector::CallContract = selector {
```

Comment:
```suggestion
        if selector == SyscallSelector::CallContract {
```

Reply:
Could we maybe add `else if`?  I guess this only makes sense if address is not cheated, it is less confusing this way and we do not have to look up a couple hundreds lines of code